internal/venice-common/src/main/java/com/linkedin/venice/servicediscovery/ServiceDiscoveryAnnouncerRetryTask.java

```diff
@@ -0,0 +1,48 @@
+package com.linkedin.venice.servicediscovery;
+
+import java.util.concurrent.BlockingQueue;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+
+/**
+ * This class is responsible for retrying the registration of a {@link ServiceDiscoveryAnnouncer} in the case of registration failure.
+ * It is a runnable task that is scheduled to register a service discovery announcer every
+ * {@link ServiceDiscoveryAnnouncerRetryTask#serviceDiscoveryRegistrationRetryMS} milliseconds.
+ */
+public class ServiceDiscoveryAnnouncerRetryTask implements Runnable {
+  private static final Logger LOGGER = LogManager.getLogger(ServiceDiscoveryAnnouncerRetryTask.class);
+  private final BlockingQueue<ServiceDiscoveryAnnouncer> serviceDiscoveryAnnouncerRetryQueue;
+  private final long serviceDiscoveryRegistrationRetryMS;
+
+  public ServiceDiscoveryAnnouncerRetryTask(
+      BlockingQueue<ServiceDiscoveryAnnouncer> serviceDiscoveryAnnouncerRetryQueue,
+      long serviceDiscoveryRegistrationRetryMS) {
+    this.serviceDiscoveryAnnouncerRetryQueue = serviceDiscoveryAnnouncerRetryQueue;
+    this.serviceDiscoveryRegistrationRetryMS = serviceDiscoveryRegistrationRetryMS;
+  }
+
+  @Override
+  public void run() {
+    while (true) {
+      ServiceDiscoveryAnnouncer announcer = null;
+      try {
+        LOGGER.info("Thread for retry task sleeping for {} ms", serviceDiscoveryRegistrationRetryMS);
+        Thread.sleep(serviceDiscoveryRegistrationRetryMS);
+        LOGGER.info("Thread for retry task woke up");
```

Comment:
Let's remove some verbosity from this logging. I don't think we need these logging.

Also, we should put the sleep after the `take` since these are two different threads and `.take` is a blocking call.